pkg/kubectl/cmd/run.go

```diff
@@ -135,6 +138,13 @@ func Run(f *cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *cob
 		return cmdutil.UsageError(cmd, "NAME is required for run")
 	}
 
+	// validate image name
+	imageName := cmdutil.GetFlagString(cmd, "image")
+	validImageRef := reference.ReferenceRegexp.FindStringSubmatch(imageName)
```

Comment:
Ah sorry, hadn't seen this comment. It would definitely make more sense. I have updated this line